src/sentry/api/endpoints/project_details.py

```diff
@@ -771,6 +778,36 @@ def delete(self, request: Request, project) -> Response:
 
         return Response(status=204)
 
+    @staticmethod
+    def _is_uniform_sampling_rule(rule):
+        # Currently UI sends top level condition op as always 'and' so asserting this here to
+        # guarantee that it will always have an `inner` key
+        assert rule["condition"]["op"] == "and"
+        # Matching the uniform sampling rule check on UI
+        return len(rule["condition"]["inner"]) == 0 and rule["type"] == "trace"
+
+    def validate_uniform_sampling_rule(self, rules):
+        # Guards against deletion of uniform sampling rule i.e. sending a payload with no rules
+        if len(rules) == 0:
+            raise UniformDynamicSamplingRuleException(
+                "Payload must contain a uniform dynamic sampling rule"
+            )
+
+        uniform_rule = rules[-1]
+        # Guards against placing uniform sampling rule not in last position or adding multiple
+        # uniform sampling rules
+        for i in range(0, len(rules) - 2):
```

Comment:
_nit_: This could be written as:

```suggestion
        for rule in rules[:-1]:
```

Reply:
Nice, added